client/components/LearnerViewDirective.js

```diff
@@ -492,7 +528,18 @@ tie.directive('learnerView', [function() {
             }
             $scope.feedbackParagraphs = congratulatoryFeedback.getParagraphs();
           } else {
-            $scope.feedbackParagraphs = feedback.getParagraphs();
+            var feedbackParagraphs = feedback.getParagraphs();
+            // Get the index of syntax error in feedback.
+            var syntaxErrorIndex = feedback.getSyntaxErrorIndex();
+            // The index must be either null(indicating no syntax error)
+            // or a positive integer.
+            if (syntaxErrorIndex) {
```

Comment:
In general, I think it's good to be more specific. E.g., if (typeof syntaxErrorIndex == 'number' && syntaxErrorIndex > 0)